### ISSUE TITLE
Guided backprop when first layer is batch normalization

### ISSUE THREAD

**Issue by deepakanandece** [utkuozbulak/pytorch-cnn-visualizations#41]
I am not able to reach the first layer in guided backprop when the first layer is batch normalization. The output size comes out to be 1-D array instead of 2-D array.

**Comment by utkuozbulak** [comment]
Hey, I am not sure why you have that problem. As far as I remember they do work with batch norm layers. (I remember getting them to work on ResNet, and it has batchnorm)

**Comment by deepakanandece** [comment]
The issue is that I am using batchnorm 1d, my input is B x C x L, so is output. (C=438, L>1)
I expect a gradient of same size
But what I get is grad_in = (None, 438 dim array, 438 dim array) tuple in
    def hook_layers(self):
        def hook_function(module, grad_in, grad_out):

**Comment by utkuozbulak** [comment]
Hello again,

As far as I remember the code expects an input of shape B(1)xCxHxW, and since you have one less dimension you don't get right shape of output. You can probably edit hook functions and whatnot to fix it for your case.

**Comment by shreygadiya** [comment]
Hi, I am working with the op, deepakanandece. As per https://pytorch.org/docs/stable/nn.html#, 
For BatchNorm1D
Input: \((N, C)\) or \((N, C, L)\)
Output: \((N, C)\) or \((N, C, L)\) (same shape as input)
To be clear, we are not dealing with images.
So I don't think dimension is the issue.
![Screenshot from 2019-04-04 18-26-44](https://user-images.githubusercontent.com/11920830/55557468-7b81c300-5707-11e9-8754-c23221a17d7c.png)

**Comment by utkuozbulak** [comment]
Hello again,

I have not tried these approaches on non-image data although I heard some colleagues were able to succeed in applying them. You should debug your code step by step and solve the problem by yourself, I'm sorry for not being able to help.

Drop a comment when you solve it so that others can benefit.
